packages/odyssey-react-mui/src/labs/DataTable.tsx

```diff
@@ -383,6 +389,132 @@ const DataTable = ({
   const rowVirtualizerInstanceRef =
     useRef<MRT_Virtualizer<HTMLDivElement, HTMLTableRowElement>>(null);
 
+  const getRowFromTableAndSetHovered = (
+    table: TableType,
+    id: MRT_RowData["id"]
+  ) => {
+    const nextRow: MRT_RowData = table.getRow(id);
+
+    if (nextRow) {
+      table.setHoveredRow(nextRow);
+    }
+  };
+
+  const resetDraggingAndHoveredRow = (table: TableType) => {
+    setDraggingRow(null);
+    table.setHoveredRow(null);
+  };
+
+  type HandleDragHandleKeyDownArgs = {
+    table: TableType;
+    row: MRT_Row<MRT_RowData>;
+    event: KeyboardEvent<HTMLButtonElement>;
+  };
+
+  const handleDragHandleKeyDown = useCallback(
+    ({ table, row, event }: HandleDragHandleKeyDownArgs) => {
+      const { hoveredRow } = table.getState();
+
+      const { key } = event;
+
+      const isSpaceKey = key === " ";
+      const isEnterKey = key === "Enter";
+      const isEscapeKey = key === "Escape";
+      const isArrowDown = key === "ArrowDown";
+      const isArrowUp = key === "ArrowUp";
+      const isSpaceOrEnter = isSpaceKey || isEnterKey;
+      const zeroIndexedPageNumber = page - 1;
+      const currentIndex = row.index + zeroIndexedPageNumber * resultsPerPage;
+
+      if (isEscapeKey) {
+        resetDraggingAndHoveredRow(table);
+        return;
+      }
+
+      if (isSpaceOrEnter) {
+        event.preventDefault();
+        event.stopPropagation();
+      }
+
+      if (draggingRow) {
+        if (typeof hoveredRow?.index === "number") {
+          const { index } = hoveredRow;
+
+          if (isSpaceOrEnter) {
+            const indexByPage =
+              zeroIndexedPageNumber > 0
+                ? index + zeroIndexedPageNumber * resultsPerPage
+                : index;
```

Comment:
I think we can do this logic without a boolean check.

Does this work?

```suggestion
            const pageRelativeIndex =
              index + (zeroIndexedPageNumber * resultsPerPage)
```

I also renamed `indexByPage`, which sounds like a sorting function, to `pageRelativeIndex` which says that this is a type of index variable.

Reply:
I like the variable name change. I'd have to try this out but, feels like it would work just fine. Thanks for the feedback

Reply:
@KevinGhadyani-Okta This works. Changed [here](https://github.com/okta/odyssey/pull/2070/commits/50e72a0b98130cbd81da3158ce476a4d88b782b6)